spec/controllers/vm_infra_controller_spec.rb

```diff
@@ -70,6 +71,256 @@
+    expect(response.status).to eq(200)
+    expect(response).to render_template(:partial => 'shared/views/_ownership')
+  end
+
+  it 'Can Extract Running Processes form VM' do
+    post :x_button, :pressed => 'vm_collect_running_processes', :id => vm_vmware.id
+    expect(response.status).to eq(200)
+  end
+
+  it 'Can Perform VM Smart State Analysis' do
+    post :x_button, :pressed => 'vm_scan', :id => vm_vmware.id
+    expect(response.status).to eq(200)
+  end
+
+  it 'Can Refresh Relationships and Power Status VM Smart State Analysis' do
+    post :x_button, :pressed => 'vm_refresh', :id => vm_vmware.id
+    expect(response.status).to eq(200)
+  end
+
+#  it 'Can open VM Compare tab' do
+#    vm = FactoryGirl.create(:vm_vmware, :host => host_1x1,
+#                            :hardware => FactoryGirl.create(:hardware, :cpu1x1, :ram1GB, :virtual_hw_version => '04'))
+#    vm2 = FactoryGirl.create(:vm_vmware, :host => host_1x1,
+#                            :hardware => FactoryGirl.create(:hardware, :cpu1x1, :ram1GB, :virtual_hw_version => '05'))
+#    FactoryGirl.create(:miq_report, :filename => 'vms.yaml',
+#                                          :template_type => 'compare',
+#                                          :name      => "Test Report",
+#                                          :rpt_type  => "Custom",
+#                                          :tz        => "Eastern Time (US & Canada)",
+#                                          :headers   => ["Name", "Boot time", "Disks aligned"],
+#                                          :col_order => ["name", "boot_time", "disks_aligned"],
+#                                          :cols      => ["name", "boot_time", "disks_aligned"])
+#
+#    #binding.pry
+#    #controller.instance_variable_set(@settings,nil)
+#    #   no implicit conversion of nil into String
+#    # ./app/helpers/application_helper/toolbar_builder.rb:1330:in `end_with?'
+#
+#    #ToolbarBuilder.stub(:build_toolbar_select_button).and_return(true)
+#    controller.stub(:x_node).and_return("v-#{vm_vmware.compressed_id}")
+#    #get :show, :id => vm_vmware.id
+#    #expect(response).to redirect_to(:action => 'explorer')
+#
+#    #post :explorer
+#    #expect(response.status).to eq(200)
+#    controller.instance_variable_set(:@settings, { :views => { :compare => "compressed" } })
+#
+#    post :x_button, :pressed => 'vm_compare', "check_#{to_cid(vm.id)}" => "1",
+#                                              "check_#{to_cid(vm2.id)}" => "1", "type" => "compressed"
+#    expect(response.status).to eq(200)
+#    #expect(response).to render_template(:partial => 'layouts/_compare')
+#  end
+
+  it 'Can open Manage Policies' do
+    #get :show, :id => vm_vmware.id
+    #expect(response).to redirect_to(:action => 'explorer')
+
+    post :explorer
+    expect(response.status).to eq(200)
+
+    post :x_button, :pressed => 'vm_protect', :id => vm_vmware.id
+    expect(response.status).to eq(200)
+  end
+
+  before do
```

Comment:
This before block is hanging out, so will run for *all* of the `it` blocks at this scoping.  This could be part of your issue.  Should it only be run for specific `it`s?  If so, those should be grouped into a context.